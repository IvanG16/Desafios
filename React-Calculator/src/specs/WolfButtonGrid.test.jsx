import WolfButtonGrid from '../components/WolfButtonGrid'
import { describe, it, expect, vi, afterEach } from 'vitest'
import { fireEvent, render, cleanup } from '@testing-library/react'

/**
 * @vitest-environment jsdom
 */

describe(' <WolfButtonGrid />', ()=> {
    const handleOnClick = vi.fn();

    afterEach(cleanup);

    it('renders properly', ()=> {
        const { container } = render(<WolfButtonGrid handleClick={handleOnClick}/>);
        expect(container).toMatchSnapshot()
    })

    it('outputs properly ', ()=> {
        const { getByText } = render(<WolfButtonGrid handleOnClick={handleOnClick}/>)
        const button = getByText('1')

        fireEvent.click(button)

        expect(handleOnClick).toHaveBeenCalledWith(1)
    })
})
