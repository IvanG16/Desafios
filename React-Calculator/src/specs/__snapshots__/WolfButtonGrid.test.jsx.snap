// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` <WolfButtonGrid /> > renders properly 1`] = `
<div>
  <div
    class="w-[425px] h-[373px] px-2.5 py-[30px] justify-start items-start gap-[25px] inline-flex items-center"
  >
    <div
      class="justify-start items-start gap-[25px] inline-flex flex-wrap items-center"
    >
      <div
        class="justify-start items-start gap-[25px] inline-flex items-center"
      >
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-50 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            Off
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-50 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            C
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-50 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            del
          </div>
        </button>
      </div>
      <div
        class="justify-start items-start gap-[25px] inline-flex flex-wrap items-center"
      >
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            7
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            8
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            9
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            4
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            5
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            6
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            1
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            2
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            3
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            .
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            0
          </div>
        </button>
        <button
          class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
          type="button"
        >
          <div
            class="w-1/2 text-black text-2xl font-normal Inconsolata"
          >
            =
          </div>
        </button>
      </div>
    </div>
    <div
      class="justify-start items-start gap-[25px] inline-flex flex-wrap items-center"
    >
      <button
        class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-50 "
        type="button"
      >
        <div
          class="w-1/2 text-black text-2xl font-normal Inconsolata"
        >
          âˆš
        </div>
      </button>
      <button
        class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-50 "
        type="button"
      >
        <div
          class="w-1/2 text-black text-2xl font-normal Inconsolata"
        >
          ^
        </div>
      </button>
      <button
        class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-50 "
        type="button"
      >
        <div
          class="w-1/2 text-black text-2xl font-normal Inconsolata"
        >
          cos
        </div>
      </button>
      <button
        class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-50 "
        type="button"
      >
        <div
          class="w-1/2 text-black text-2xl font-normal Inconsolata"
        >
          tan
        </div>
      </button>
      <button
        class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-50 "
        type="button"
      >
        <div
          class="w-1/2 text-black text-2xl font-normal Inconsolata"
        >
          sen
        </div>
      </button>
    </div>
  </div>
</div>
`;
